### PR TITLE
Add Nurse class for MediSync

### LINKED ISSUE #21: As a head nurse, I can view the patients whom the nurses are in charge of
... so that I can balance the workload of my nurses

### LINKED ISSUE #104: Implement Nurse class

### PR DESCRIPTION
**Pull request by leezhanpeng**
Main Changes:
- `Nurse` class is now available.
- `Staff` class is introduced as a `Person` that contains `shiftDays` attribute.
- `Doctor` and `Nurse` class extends from `Staff` class.
- Ui now displays the email of the `Person`, and also formatted to include `Nurse`.
- `SampleDataUtil` is now updated to include `Nurse`s when a new data initialisation happens for MediSync.

Bug fixes:
- Any edits to `Staff` will erase its `shiftDays`. `EditCommand`'s `execute()` has been adjusted to fix this.

### CODE REVIEW

**Comment by leezhanpeng** [comment]
As this PR contains edits to Ui codes, it is unable to pass the patch test in codecov.

File: src/main/java/seedu/address/storage/CustomJsonAdaptedPersonSerializer.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  *
  * <p>The serialization process includes converting fields such as name, phone, email, role, affiliations,
  * and affiliation history to JSON fields. Additionally, it conditionally includes the shiftDays
- * field in the JSON output based on the person's role. If the role is "Doctor", the shiftDays field is
+ * field in the JSON output based on the person's role. If the role is "Staff", the shiftDays field is
```
Comment (shuenj) [inline src/main/java/seedu/address/storage/CustomJsonAdaptedPersonSerializer.java]: Just to clarify, we will not have a role "Staff" right, it is sort of an abstract class for Doctor and Nurse?

**Review by shuenj — APPROVED** [review]
LGTM, nice work!

File: src/main/java/seedu/address/storage/CustomJsonAdaptedPersonSerializer.java (hunk 1 above)
Comment (leezhanpeng) [reply src/main/java/seedu/address/storage/CustomJsonAdaptedPersonSerializer.java]: That's right. However for the sake of testability and sanity, I have not made it abstract.

Will modify the comment for better clarity!

Comment (shuenj) [reply src/main/java/seedu/address/storage/CustomJsonAdaptedPersonSerializer.java]: Alright, understood, thanks!

**Pull request merged**